### PR TITLE
Update External OAuth Providers

### PR DESCRIPTION
**Pull request by HarryET**
## What kind of change does this PR introduce?

Adds support for new external OAuth providers (discord, twitter & apple)

## What is the current behavior?

No support for specific 0Auth providers

## What is the new behavior?

Supports the new 0Auth providers

## Additional context

N/a

### CODE REVIEW

File: lib/src/provider.dart [hunk 1]
```diff
@@ -1,4 +1,4 @@
-enum Provider { azure, bitbucket, facebook, github, gitlab, google }
+enum Provider { apple, azure, bitbucket, discord, facebook, github, gitlab, google, twitter }
```
Comment (dshukertjr) [inline lib/src/provider.dart:1]: Appreciate that you sorted these in alphabetical order! There seems to be a formatting error that can be fixed by replacing this line with this:

```suggestion
enum Provider {
  apple,
  azure,
  bitbucket,
  discord,
  facebook,
  github,
  gitlab,
  google,
  twitter
}
```

**Comment by dshukertjr** [comment]
Thank you so much @HarryET for your amazing work!  There was a small formatting issue, but other than that, LGTM!

**Comment by HarryET** [comment]
Ah, sorry! Did it quickly. Supabase looks awesome and can't wait to see how I can help and work with it in the future 😀

**Comment by dshukertjr** [comment]
I fixed the formatting, but tests are failing now, and I'm pretty sure I have screwed up something 😅 Trying to figure out why tests are failing right now...

**Comment by dshukertjr** [comment]
I reran a [past github action flow that was passing](https://github.com/supabase/gotrue-dart/actions/runs/991729244), and it failed. It seems like this issue might take some deep dive to solve. @phamhieu are you aware of anything that might have affected this test result?

**Comment by HarryET** [comment]
I know discord auth is working as I have tested that but unsure about the other's status as per working

**Comment by dshukertjr** [comment]
Yup, the tests failing seems to be unrelated to the code change introduced in this PR, which is weird ...

**Comment by phamhieu** [comment]
lgtm, i will debug unit test as a separate issue.

**Pull request merged**